server/src/main/java/org/candlepin/pki/impl/JSSPKIUtility.java

```diff
@@ -0,0 +1,354 @@
+
+            // The subject key identifier of the CA becomes the Authority Key Identifer of the CRL.
+            byte[] extValue = caCert.getExtensionValue(PKIXExtensions.SubjectKey_Id.toString());
+
+            // The getExtensionValue returns us the Extension extnValue element which is an octet string.  The
+            // actual KeyIdentifier is also an octet string.  The extnValue for the SubjectKeyIdentifier
+            // extension is an octet string of type KeyIdentifier and therefore ultimately an octet string
+            // of an octet string.  See Appendix A of RFC 5280.
+            OCTET_STRING extOctets = (OCTET_STRING) ASN1Util.decode(new OCTET_STRING.Template(), extValue);
+            OCTET_STRING ski = (OCTET_STRING) ASN1Util.decode(new OCTET_STRING.Template(), extOctets
+                .toByteArray());
+
+            if (ski == null) {
+                /* RFC 5280 mandates that this extension be present but sometimes it isn't.  If it's not
+                 * we could still calculate the subject key identifier to use by taking the SHA1 digest
+                 * of the public key, but that's harder than it sounds with JSS and all to support
+                 * a cert that is malformed (plus it would require creating a malformed cert to use
+                 * for testing), so I'm going to leave this unfinished until we really need it. */
+                throw new IOException("Could not find SubjectKeyIdentifier extention in CA");
+            }
+
+            KeyIdentifier ki = new KeyIdentifier(ski.toByteArray());
+
+            GeneralName[] akiName = new GeneralName[1];
+            akiName[0] = new GeneralName(new X500Name(caCert.getIssuerX500Principal().getEncoded()));
+            GeneralNames generalNames = new GeneralNames(akiName);
+
+            SerialNumber serialNumber = new SerialNumber(caCert.getSerialNumber());
+
+            crlExtensions.add(new AuthorityKeyIdentifierExtension(ki, generalNames, serialNumber));
+
+            X500Name issuer = new X500Name(caCert.getIssuerX500Principal().getEncoded());
+            Date until = Util.addDaysToDt(config.getInt(ConfigProperties.CRL_NEXT_UPDATE_DELTA));
+            X509CRLImpl crlImpl = new X509CRLImpl(
+                issuer,
+                new Date(),
+                until,
+                revokedCerts.toArray(new RevokedCertificate[] {}),
+                crlExtensions
+            );
+
+            crlImpl.sign(reader.getCaKey(), SIGNING_ALG_ID);
+
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            return (X509CRL) cf.generateCRL(new ByteArrayInputStream(crlImpl.getEncoded()));
+        }
+        catch (GeneralSecurityException | IOException | InvalidBERException e) {
+            throw new RuntimeException("Error creating CRL", e);
+        }
+    }
+
+    private byte[] getPemEncoded(byte[] der, String type) throws IOException {
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
```

Comment:
OT comment:
Does our linter handle this very well? I can see this being confused for an inline assignment.

Reply:
It doesn't complain at least.  We have the option of doing

```java
try (
  OutputStream x = new ByteArrayOutputStream();
) {
```

I can change it to that if you want.

Reply:
Not at all -- I don't mind the new try-with-resources syntax, I just expected our linter to be a little flaky with a "new" syntax that looks similar to old disallowed ones.